cmd/ci-operator/main.go

```diff
@@ -870,6 +870,30 @@ func (o *options) initializeNamespace() error {
 		return errors.New("timed out waiting for RBAC")
 	}
 
+	var imagePullSecretsMinted bool
+	for i := 0; i < 30; i++ {
+		imagePullSecretsMinted = true
+		serviceAccounts := map[string]*coreapi.ServiceAccount{
+			"builder": {},
+			"default": {},
+		}
+		for name := range serviceAccounts {
+			if err := client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: o.namespace, Name: name}, serviceAccounts[name]); err != nil {
+				return fmt.Errorf("failed to fetch service account %s: %w", name, err)
+			}
+			imagePullSecretsMinted = imagePullSecretsMinted && len(serviceAccounts[name].ImagePullSecrets) > 0
+		}
+		if imagePullSecretsMinted {
+			break
+		}
+		log.Printf("[%d/30] Image pull secrets in namespace not yet ready, sleeping for a second...\n", i)
+		time.Sleep(time.Second)
+	}
```

Comment:
Nit: To satisfy my curiosity, can we add a "Waited for pull secrets: n seconds" logline that is easily searchable in ci-search so we can get an impression of how long this usually takes?